File: MyTestVueApp.Server/Entities/Art.cs
using System.Reflection.Metadata.Ecma335;

namespace MyTestVueApp.Server.Entities
{
    public class Art
    {
        //Required
        public int Id { get; set; }
        public int[] ArtistId { get; set; }
        public string Title { get; set; }
        public bool IsPublic { get; set; }
        public DateTime CreationDate { get; set; }
        public PixelGrid PixelGrid { get; set; }


        //Optional external values
        public string[] ArtistName { get; set; }
        public bool IsGif { get; set; }
        public int GifID { get; set; }
        public int GifFrameNum { get; set; }
        public int GifFps { get; set; }
        public int NumLikes { get; set; }
        public int NumComments { get; set; }
        public bool CurrentUserIsOwner { get; set; } = false;


        public void SetArtists(List<Artist> artists)
        {
            ArtistId = artists.Select((artist) => artist.Id).ToArray();
            ArtistName = artists.Select(artist => artist.Name).ToArray();
        }
    }
}

File: mytestvueapp.client/src/entities/Art.ts
import { PixelGrid } from "./PixelGrid";

export default class Art {
  //required
  id: number;
  artistId: number[];
  title: string;
  isPublic: boolean;
  creationDate: string;

  //may be null if new
  pixelGrid: PixelGrid;

  //optional
  artistName: string[];
  numLikes: number;
  numComments: number;
  currentUserIsOwner: boolean;
  isLiked: boolean;

  isGif: boolean;
  gifID: number;
  gifFrameNum: number;
  gifFps: number;

  constructor() {
    this.id = 0;
    this.title = "";
    this.artistId = [];
    this.artistName = [];

    this.creationDate = "";
    this.isPublic = false;
    this.numLikes = 0;
    this.numComments = 0;
    this.pixelGrid = new PixelGrid(1, 1, "FF0000", false);
    this.currentUserIsOwner = false;

    this.isGif = false;
    this.gifFrameNum = 0;
    this.gifID = 0;
    this.gifFps = 0;
    this.isLiked = false;
  }
}
